### PR TITLE
travis.yml to manage dependencies using Go modules

### PR DESCRIPTION
**Pull request by exfly**
#183

### CODE REVIEW

**Comment by teivah** [comment]
Hey @exfly, thank you for your PR. Sorry but could you please apply this PR to the `v2` branch? I do not want to test the `v1` except for bug fixes.
For information, each issue is tagged with the target version.
Thanks!

**Comment by teivah** [comment]
@exfly I meant to merge your changes in the `v2` branch, not to merge the changes from `v2` to `master`. The branch is not ready yet.

**Comment by exfly** [comment]
@teivah sorr～now？

**Comment by exfly** [comment]
@venth Can we give up the support of  gov1.11-?

**Comment by teivah** [comment]
@exfly I _fixed_ the failing tests, can you please merge again from `v2` branch? Should be good now.

**Comment by exfly** [comment]
@teivah thanks. In addition, I have reorganized the new commit.

**Comment by teivah** [comment]
@exfly Looks good now. I'm gonna take a deep look at the PR as soon as I can. Thank you!

File: .gitignore [hunk 1]
```diff
@@ -34,3 +34,5 @@ node_modules
 # IntelliJ
 .idea
 *.iml
+
+go.sum
```
Comment (teivah) [inline .gitignore]: I'm more in favor of keeping `go.sum` as described here: https://github.com/golang/go/wiki/Modules#should-i-commit-my-gosum-file-as-well-as-my-gomod-file

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,53 @@
+export GO111MODULE=on
+export GOPROXY=https://goproxy.io
```
Comment (teivah) [inline Makefile]: What's the goal of this proxy?

File: .travis.yml [hunk 3]
```diff
@@ -4,8 +4,12 @@ go:
   - 1.9
   - '1.10'
   - '1.11'
+  - '1.12'
   - master
 
+env:
```
Comment (teivah) [inline .travis.yml]: Why do we have still those `go get` in the Travis file then?

Comment (exfly) [reply .travis.yml]: Whether or not gomod is installed using go get

File: Makefile (hunk 2 above)
Comment (exfly) [reply Makefile]: Speed up download dependencies. Works on go mod

**Comment by exfly** [comment]
@teivah When can this pr merge to v2?

File: Makefile (hunk 2 above)
Comment (teivah) [reply Makefile]: Do you think we could consider switching to https://proxy.golang.org?

File: .travis.yml (hunk 3 above)
Comment (teivah) [reply .travis.yml]: I don't understand. Why do we have these?
```
install:
  - go get github.com/mattn/goveralls
  - go get github.com/stretchr/testify/assert
  - go get github.com/gorilla/websocket
  - go get github.com/onsi/ginkgo
  - go get github.com/onsi/gomega
  - go get golang.org/x/sync/semaphore
```
If we use go mod, what's the point of keeping manual go get like this?

**Comment by exfly** [comment]
I want to be compatible with the lower version of go, but it seems that this method is not very good. Wait minitue, I will fix it.

**Comment by exfly** [comment]
I think, in order to be compatible with the lower version of go,we still need to get the dependency with go get. For the high version of go, `go get` can `get the uses the network to check out
missing packages but does not use it to look for updates to existing packages`.

**Comment by exfly** [comment]
@teivah anything can I improve this pr?

**Comment by teivah** [comment]
Implemented in v2 branch

**Pull request closed without merging**